hphp/runtime/vm/jit/unwind-itanium.cpp

```diff
@@ -358,7 +365,24 @@ void write_tc_cie(EHFrameWriter& ehfw) {
 
   // The following calculation is related to the "top" of the record.  There is
   // an implicit -8 factor as defined by EHFrameWriter::m_cie.data_align.
+
+#if defined(__powerpc64__)
```

Comment:
@mxw This was the responsible for all the issues we had with frames. As I have already told you, a execution stores the return address in the parent's frame, so the current frame have an invalid information in the place of its m_savedRip, which will be used in the next call.
When unwinding, HHVM finds out if a frame is a VM frame analyzing the m_savedRip, which must be the one from the current context, so the following rules get it from the parent frame.
This was the reason the unwind was trying to unwind an addition PHP frame before get the last C++ frame, the m_savedRip being considered was from the next frame, i. e., still a PHP frame.